pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -696,11 +736,6 @@ pub mod pallet {
 			)?
 			.into();
 
-			if burnt.is_zero() {
-				// no funds available to be burnt on account; nothing to do here
-				return Ok(());
-			}
-
```

Comment:
@rflechtner is it a breaking change, if I drop this if statement? Otherwise, I need a special treatment for those early returns in the benchmarking

Reply:
it is indeed a breaking change, because this will lead to errors. But why don't you just use a weight like `Ok(Some(T::WeightInfo::refund_account(0)).into())` given that the amount of currencies up to this point does not matter at all?

Reply:
the one below will also cause errors when removed, as per the comment included. Any reason you can't just use a weight of `Ok(Some(T::WeightInfo::refund_account(currency_count.to_owned())).into())` like for the final result? If we don't cover these cases in benchmarking at all (so we make sure there's always a balance and a collateral returned), we would simply overestimate the weight here, which shouldn't be an issue?

Reply:
> it is indeed a breaking change, because this will lead to errors

Well actually it's already broken now because trying to unfreeze an account that does not exist will error. So we need to decide if trying to refund an account that does not exist should be treated as a no-op, which means fixing this, or if it should be treated as an error. Ultimately both work, you could just submit a batchAll if you wanted to ignore that some accounts in your list may not have needed refunding.

Reply:
I think, trying to refund an account that does not exist should throw an error. Feels semantically cleaner to me. But I have no strong opinion on that.

Reply:
I'm also fine with it. You still have a way to do fire and forget if you opt for a batchAll

Reply:
Added in: [2bced52](https://github.com/KILTprotocol/kilt-node/pull/789/commits/2bced5208c970b38a109a4c74063769d82407e16)

Reply:
🤘